src/charm.py

```diff
@@ -95,6 +114,36 @@ def _get_pebble_config(self, event: ops.framework.EventBase) -> dict:
             pebble_config["services"]["gunicorn"]["environment"] = pod_env_config
         return pebble_config
 
+    def _get_statsd_pebble_config(self, event: ops.framework.EventBase) -> dict:
+        """Generate statsd exporter pebble config."""
+        pebble_config = {
+            "summary": "gunicorn layer",
+            "description": "gunicorn layer",
+            "services": {
+                "statsd-prometheus-exporter": {
+                    "override": "replace",
+                    "summary": "statsd exporter service",
+                    "user": "nobody",
+                    "command": "/bin/statsd_exporter",
+                    "startup": "enabled",
+                }
+            },
+            "checks": {
+                "container-ready": {
+                    "override": "replace",
+                    "level": "ready",
+                    "http": {"url": "http://localhost:9102/metrics"},
+                },
+            },
+        }
+
+        juju_conf = self._check_juju_config()
+        if juju_conf:
+            self.unit.status = BlockedStatus(str(juju_conf))
+            return {}
```

Comment:
I don't think we should be doing this within this function. We're duplicating something that's being done in another function (_get_pebble_config). I think `juju_conf = self._check_juju_config()` should be pulled out of `_get_pebble_config`, and in `_configure_workload` we should get our pebble configs for each container, and then do the juju config check separately.